src/test/lib/statistics/statistics_objects/min_max_filter_test.cpp

```diff
@@ -48,6 +48,29 @@ class MinMaxFilterTest<pmr_string> : public BaseTest {
   pmr_string _before_range, _min_value, _max_value, _after_range, _in_between, _in_between2;
 };
 
+class MinMaxFilterTestLike : public BaseTest {
+ protected:
+  void SetUp() override { _values = pmr_vector<pmr_string>{"bb", "cc", "dd"}; }
+  pmr_vector<pmr_string> _values;
+};
+
+TEST_F(MinMaxFilterTestLike, CanPrune) {
+  auto filter = std::make_unique<MinMaxFilter<pmr_string>>(this->_values.front(), this->_values.back());
+  // for the predicate condition of Like, we expect only values where the lower_bound is bigger then max
+  // or the upper_bound is smaller then min to be prunable
+
+  EXPECT_TRUE(filter->does_not_contain(PredicateCondition::Like, "aa%"));
+  EXPECT_TRUE(filter->does_not_contain(PredicateCondition::Like, "aa_"));
+  EXPECT_TRUE(filter->does_not_contain(PredicateCondition::Like, "ee%"));
+  EXPECT_TRUE(filter->does_not_contain(PredicateCondition::Like, "ee_"));
+
+  EXPECT_FALSE(filter->does_not_contain(PredicateCondition::Like, "ba%"));
+  EXPECT_FALSE(filter->does_not_contain(PredicateCondition::Like, "dd%"));
+  EXPECT_FALSE(filter->does_not_contain(PredicateCondition::Like, "cc%"));
+  EXPECT_FALSE(filter->does_not_contain(PredicateCondition::Like, "%"));
```

Comment:
What about multiple wildcards and about a longer search word with a wild card in the beginning?

I'd also test the min/max (you have tested `dd`, but not `bb`) and single chars with wildcards.

Tbh, I was not even aware of the existince of the `_` wildcard. Would a test of `a_%` make sense?

Reply:
I will adjust the tests and a test with`a_%` would definitely make sense. In this case, we would test that `_` is the first wildcard which is used to calculate the upper bound. Thanks for the suggestion 🙃 .